homeassistant/components/wsdot/config_flow.py

```diff
@@ -0,0 +1,157 @@
+"""Adds config flow for wsdot."""
+
+import logging
+from types import MappingProxyType
+from typing import Any
+
+from aiohttp.client_exceptions import ClientError
+import voluptuous as vol
+import wsdot as wsdot_api
+
+from homeassistant.config_entries import (
+    SOURCE_USER,
+    ConfigEntry,
+    ConfigFlow,
+    ConfigFlowResult,
+    ConfigSubentry,
+    ConfigSubentryFlow,
+    SubentryFlowResult,
+)
+from homeassistant.const import CONF_API_KEY, CONF_ID, CONF_NAME
+from homeassistant.core import callback
+from homeassistant.helpers.selector import SelectSelector, SelectSelectorConfig
+
+from .const import CONF_TRAVEL_TIMES, DOMAIN, SUBENTRY_TRAVEL_TIMES
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class InvalidApiKeyError(ClientError):
+    """Exception indicating the user entered an invalid API Key."""
+
+
+class WSDOTConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Config flow for WSDOT."""
+
+    VERSION = 1
+    MINOR_VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            wsdot_travel_times = wsdot_api.WsdotTravelTimes(user_input[CONF_API_KEY])
+            try:
+                await wsdot_travel_times.get_all_travel_times()
+            except wsdot_api.WsdotTravelError as ws_error:
+                if ws_error.status == 400:
+                    errors[CONF_API_KEY] = "invalid_api_key"
+                else:
+                    errors["base"] = "cannot_connect"
+            else:
+                await self.async_set_unique_id(user_input[CONF_API_KEY])
+                data = {CONF_API_KEY: user_input[CONF_API_KEY]}
+                self._async_abort_entries_match(data)
```

Comment:
```suggestion
                await self.async_set_unique_id(user_input[CONF_API_KEY])
                self._abort_if_unique_id_configured()
```

Reply:
This directly contradicts https://github.com/home-assistant/core/pull/149208#discussion_r2392578694?